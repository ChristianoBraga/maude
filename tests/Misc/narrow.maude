set show timing off .

mod BAZ is
  sort Foo .
  ops f g h : Foo -> Foo .
  ops i j k : Foo Foo -> Foo .

vars X Y Z W A B C D : Foo .
  eq j(f(X), Y) = i(f(Y), X) [variant] .
  rl g(i(X, k(Y, Z))) => f(k(Z, X)) .
  rl f(k(X, X)) => h(X) .
endm

set show breakdown on .

vu-narrow in BAZ : g(j(A, B)) =>* C .
vu-narrow in BAZ : g(j(A, B)) =>+ C .
vu-narrow in BAZ : g(j(A, B)) =>! C .

vu-narrow [,1] in BAZ : g(j(A, B)) =>! C .
vu-narrow [,2] in BAZ : g(j(A, B)) =>! C .
vu-narrow [,3] in BAZ : g(j(A, B)) =>! C .
vu-narrow [,4] in BAZ : g(j(A, B)) =>! C .

vu-narrow [,1] in BAZ : g(j(A, B)) =>* C .
vu-narrow [,2] in BAZ : g(j(A, B)) =>* C .
vu-narrow [,3] in BAZ : g(j(A, B)) =>* C .
