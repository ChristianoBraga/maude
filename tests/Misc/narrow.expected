Maude> ==========================================
vu-narrow in BAZ : g(j(A, B)) =>* C .

Solution 1
rewrites: 1
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 1  narrowing steps: 0
state: g(j(#1:Foo, #2:Foo))
C --> g(j(@1:Foo, @2:Foo))
#1:Foo --> @1:Foo
#2:Foo --> @2:Foo

Solution 2
rewrites: 1
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 1  narrowing steps: 0
state: g(j(#1:Foo, #2:Foo))
C --> g(i(f(%1:Foo), %2:Foo))
#1:Foo --> f(%2:Foo)
#2:Foo --> %1:Foo

Solution 3
rewrites: 3
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 2  narrowing steps: 1
state: f(k(%2:Foo, f(%3:Foo)))
C --> f(k(@1:Foo, f(@2:Foo)))
%2:Foo --> @1:Foo
%3:Foo --> @2:Foo

Solution 4
rewrites: 7
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 5  narrowing steps: 2
state: h(f(@1:Foo))
C --> h(f(%1:Foo))
@1:Foo --> %1:Foo

Solution 5
rewrites: 8
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 5  narrowing steps: 3
state: f(k(@2:Foo, h(@1:Foo)))
C --> f(k(%1:Foo, h(%2:Foo)))
@2:Foo --> %1:Foo
@1:Foo --> %2:Foo

Solution 6
rewrites: 9
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 5  narrowing steps: 4
state: h(h(%1:Foo))
C --> h(h(@1:Foo))
%1:Foo --> @1:Foo

Solution 7
rewrites: 10
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 5  narrowing steps: 5
state: h(h(%1:Foo))
C --> h(h(@1:Foo))
%1:Foo --> @1:Foo

No more solutions.
rewrites: 10
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 5  narrowing steps: 5
==========================================
vu-narrow in BAZ : g(j(A, B)) =>+ C .

Solution 1
rewrites: 2
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 1  narrowing steps: 1
state: f(k(%2:Foo, f(%3:Foo)))
C --> f(k(@1:Foo, f(@2:Foo)))
%2:Foo --> @1:Foo
%3:Foo --> @2:Foo

Solution 2
rewrites: 6
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 4  narrowing steps: 2
state: h(f(@1:Foo))
C --> h(f(%1:Foo))
@1:Foo --> %1:Foo

Solution 3
rewrites: 7
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 4  narrowing steps: 3
state: f(k(@2:Foo, h(@1:Foo)))
C --> f(k(%1:Foo, h(%2:Foo)))
@2:Foo --> %1:Foo
@1:Foo --> %2:Foo

Solution 4
rewrites: 8
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 4  narrowing steps: 4
state: h(h(%1:Foo))
C --> h(h(@1:Foo))
%1:Foo --> @1:Foo

Solution 5
rewrites: 9
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 4  narrowing steps: 5
state: h(h(%1:Foo))
C --> h(h(@1:Foo))
%1:Foo --> @1:Foo

No more solutions.
rewrites: 9
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 4  narrowing steps: 5
==========================================
vu-narrow in BAZ : g(j(A, B)) =>! C .

Solution 1
rewrites: 9
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 4  narrowing steps: 5
state: h(h(%1:Foo))
C --> h(h(@1:Foo))
%1:Foo --> @1:Foo

Solution 2
rewrites: 9
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 4  narrowing steps: 5
state: h(h(%1:Foo))
C --> h(h(@1:Foo))
%1:Foo --> @1:Foo

No more solutions.
rewrites: 9
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 4  narrowing steps: 5
==========================================
vu-narrow [, 1] in BAZ : g(j(A, B)) =>! C .

No solution.
rewrites: 5
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 4  narrowing steps: 1
==========================================
vu-narrow [, 2] in BAZ : g(j(A, B)) =>! C .

No solution.
rewrites: 7
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 4  narrowing steps: 3
==========================================
vu-narrow [, 3] in BAZ : g(j(A, B)) =>! C .

Solution 1
rewrites: 9
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 4  narrowing steps: 5
state: h(h(%1:Foo))
C --> h(h(@1:Foo))
%1:Foo --> @1:Foo

Solution 2
rewrites: 9
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 4  narrowing steps: 5
state: h(h(%1:Foo))
C --> h(h(@1:Foo))
%1:Foo --> @1:Foo

No more solutions.
rewrites: 9
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 4  narrowing steps: 5
==========================================
vu-narrow [, 4] in BAZ : g(j(A, B)) =>! C .

Solution 1
rewrites: 9
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 4  narrowing steps: 5
state: h(h(%1:Foo))
C --> h(h(@1:Foo))
%1:Foo --> @1:Foo

Solution 2
rewrites: 9
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 4  narrowing steps: 5
state: h(h(%1:Foo))
C --> h(h(@1:Foo))
%1:Foo --> @1:Foo

No more solutions.
rewrites: 9
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 4  narrowing steps: 5
==========================================
vu-narrow [, 1] in BAZ : g(j(A, B)) =>* C .

Solution 1
rewrites: 1
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 1  narrowing steps: 0
state: g(j(#1:Foo, #2:Foo))
C --> g(j(@1:Foo, @2:Foo))
#1:Foo --> @1:Foo
#2:Foo --> @2:Foo

Solution 2
rewrites: 1
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 1  narrowing steps: 0
state: g(j(#1:Foo, #2:Foo))
C --> g(i(f(%1:Foo), %2:Foo))
#1:Foo --> f(%2:Foo)
#2:Foo --> %1:Foo

Solution 3
rewrites: 3
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 2  narrowing steps: 1
state: f(k(%2:Foo, f(%3:Foo)))
C --> f(k(@1:Foo, f(@2:Foo)))
%2:Foo --> @1:Foo
%3:Foo --> @2:Foo

No more solutions.
rewrites: 6
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 5  narrowing steps: 1
==========================================
vu-narrow [, 2] in BAZ : g(j(A, B)) =>* C .

Solution 1
rewrites: 1
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 1  narrowing steps: 0
state: g(j(#1:Foo, #2:Foo))
C --> g(j(@1:Foo, @2:Foo))
#1:Foo --> @1:Foo
#2:Foo --> @2:Foo

Solution 2
rewrites: 1
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 1  narrowing steps: 0
state: g(j(#1:Foo, #2:Foo))
C --> g(i(f(%1:Foo), %2:Foo))
#1:Foo --> f(%2:Foo)
#2:Foo --> %1:Foo

Solution 3
rewrites: 3
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 2  narrowing steps: 1
state: f(k(%2:Foo, f(%3:Foo)))
C --> f(k(@1:Foo, f(@2:Foo)))
%2:Foo --> @1:Foo
%3:Foo --> @2:Foo

Solution 4
rewrites: 7
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 5  narrowing steps: 2
state: h(f(@1:Foo))
C --> h(f(%1:Foo))
@1:Foo --> %1:Foo

Solution 5
rewrites: 8
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 5  narrowing steps: 3
state: f(k(@2:Foo, h(@1:Foo)))
C --> f(k(%1:Foo, h(%2:Foo)))
@2:Foo --> %1:Foo
@1:Foo --> %2:Foo

No more solutions.
rewrites: 8
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 5  narrowing steps: 3
==========================================
vu-narrow [, 3] in BAZ : g(j(A, B)) =>* C .

Solution 1
rewrites: 1
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 1  narrowing steps: 0
state: g(j(#1:Foo, #2:Foo))
C --> g(j(@1:Foo, @2:Foo))
#1:Foo --> @1:Foo
#2:Foo --> @2:Foo

Solution 2
rewrites: 1
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 1  narrowing steps: 0
state: g(j(#1:Foo, #2:Foo))
C --> g(i(f(%1:Foo), %2:Foo))
#1:Foo --> f(%2:Foo)
#2:Foo --> %1:Foo

Solution 3
rewrites: 3
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 2  narrowing steps: 1
state: f(k(%2:Foo, f(%3:Foo)))
C --> f(k(@1:Foo, f(@2:Foo)))
%2:Foo --> @1:Foo
%3:Foo --> @2:Foo

Solution 4
rewrites: 7
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 5  narrowing steps: 2
state: h(f(@1:Foo))
C --> h(f(%1:Foo))
@1:Foo --> %1:Foo

Solution 5
rewrites: 8
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 5  narrowing steps: 3
state: f(k(@2:Foo, h(@1:Foo)))
C --> f(k(%1:Foo, h(%2:Foo)))
@2:Foo --> %1:Foo
@1:Foo --> %2:Foo

Solution 6
rewrites: 9
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 5  narrowing steps: 4
state: h(h(%1:Foo))
C --> h(h(@1:Foo))
%1:Foo --> @1:Foo

Solution 7
rewrites: 10
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 5  narrowing steps: 5
state: h(h(%1:Foo))
C --> h(h(@1:Foo))
%1:Foo --> @1:Foo

No more solutions.
rewrites: 10
mb applications: 0  equational rewrites: 0  rule rewrites: 0  variant narrowing
    steps: 5  narrowing steps: 5
Maude> Bye.
